pkg/datapath/linux/probes/probes.go

```diff
@@ -435,6 +436,41 @@ func HaveIPv6Support() error {
 	return nil
 }
 
+// HaveFibLookupSkipNeigh tests whether or not the kernel supports the
+// BPF_FIB_LOOKUP_SKIP_NEIGH flag for bpf_fib_lookup.
+// https://lore.kernel.org/bpf/20230217205515.3583372-1-martin.lau@linux.dev/
+var HaveFibLookupSkipNeigh = sync.OnceValue(func() error {
+	var objs bpfgen.ProbesObjects
+
+	if err := bpfgen.LoadProbesObjects(&objs, &ebpf.CollectionOptions{}); err != nil {
+		return fmt.Errorf("loading collection: %w", err)
+	}
+
+	defer objs.Close()
+
+	ret, err := objs.ProbeFibLookupSkipNeigh.Run(&ebpf.RunOptions{
+		// Newer kernels require that data is at least 14 bytes:
+		// https://github.com/torvalds/linux/commit/6b3d638ca897e099fa99bd6d02189d3176f80a47
+		Data:   make([]byte, 14),
+		Repeat: 1,
+	})
+	var ve *ebpf.VerifierError
```

Comment:
This doesn't look right to me. Shouldn't the check for a verifier error happen after loading the program, in this case `LoadProbesObjects`?

Reply:
Yes. Good catch. I moved some things around and forgot to adjust where I do the `VerifierError` check. Thanks!